components/multiselect/multiselect_list.jsx

```diff
@@ -50,6 +50,10 @@ export default class MultiSelectList extends React.Component {
         }
     }
 
+    setSelected = (selected) => {
```

Comment:
Couldn't figure out who was calling this till @marianunez pointed out that it was being called externally https://github.com/mattermost/mattermost-webapp/blob/517d37a22220d22b95d22dcf2a4910c8588d862e/components/multiselect/multiselect.jsx#L148

I'm afraid someone will come here and remove this function again as they'll go through the same thought process as I did of thinking that this is not being invoked at all.

@marianunez mind adding a comment here mentioning that this is being used outside of this component and adding a small test to lock down the behaviour?

Reply:
just a small test to make sure `setSelected` is being invoked or something.

Reply:
I've added a unit test to ensure that is being called but given that it is mocked it does not guarantee that the actual implementation of `setSelected` still exists. Open to other suggestions on this

Reply:
One way to test is:
```javascript
import {mountWithIntl} from 'tests/helpers/intl-test-helper.jsx';

import MultiSelectList from './multiselect_list.jsx';
...
    test('MultiSelectList should match state and call setSelected on next page', () => {
        function renderOption(option, isSelected, onAdd) {
            return (
                <p
                    key={option.id}
                    ref={isSelected ? 'selected' : option.id}
                    onClick={() => onAdd(option)}
                >
                    {option.id}
                </p>
            );
        }

        function renderValue(props) {
            return props.data.value;
        }

        const wrapper = mountWithIntl(
            <MultiSelect
                {...baseProps}
                optionRenderer={renderOption}
                valueRenderer={renderValue}
            />
        );

        expect(wrapper.ref('list')).toBeTruthy();
        expect(wrapper.find(MultiSelectList).state('selected')).toEqual(-1);

        wrapper.find('.filter-control__next').simulate('click');
        expect(wrapper.find(MultiSelectList).state('selected')).toEqual(0);

        const listRef = wrapper.ref('list');
        listRef.setSelected = jest.fn();
        wrapper.instance().nextPage();
        expect(listRef.setSelected).toHaveBeenCalledTimes(1);
    });
```

Reply:
Works like a charm @saturninoabril! I was having trouble with the `mount` and  wasn't aware we had a `mountWithIntl` helper 🎉 Thanks for this!

I've updated the test with this one 👍

Reply:
Nice test!